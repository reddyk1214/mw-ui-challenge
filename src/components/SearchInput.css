.SearchInputContainer {
  display: flex;
  align-items: center;
  background-color: rgba(255, 255, 255, 0.02);
  border-radius: 3px;
  width: 220px;
  height: 32px;
}

.SearchInput {
  flex-grow: 1;
  height: 100%;
  padding-right: 12px;
  background-color: transparent;
  border: none;
  outline: none;
  font-family: 'Roboto', sans-serif;
  font-size: 12px;
  margin-left: 8px;
  color: rgba(255, 255, 255, 0.5);
}

.SearchInput::placeholder {
  font-family: 'Roboto', sans-serif;
  color: rgba(255, 255, 255, 0.5);
}

.SearchIconContainer {
  display: flex;
  align-items: center;
  justify-content: center;
  width: 24px;
  height: 24px;
}

.SearchIcon {
  width: 16px;
  height: 16px;
}
