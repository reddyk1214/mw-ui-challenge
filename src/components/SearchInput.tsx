import './SearchInput.css';
import searchIcon from '../icons/search_16px.svg';

interface SearchInputProps {
  searchText: string;
  onSearch: (text: string) => void;
}

export function SearchInput({ searchText, onSearch }: SearchInputProps): JSX.Element | null {
  return (
    <div className="SearchInputContainer">
      <div className="SearchIconContainer">
        <img className="SearchIcon" src={searchIcon} alt="Search Icon" />
      </div>
      <input
        data-testid="search-input"
        className="SearchInput"
        type="search"
        value={searchText}
        onChange={(e) => onSearch(e.target.value)} 
        placeholder="Search"
        aria-label="Search positions"
      />
    </div>
  );
}
