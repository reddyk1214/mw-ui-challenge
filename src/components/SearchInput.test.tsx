// @vitest-environment jsdom
import { render, fireEvent, cleanup } from '@testing-library/react';
import { expect, test, describe, vi, beforeEach, afterEach } from 'vitest';
import { SearchInput } from './SearchInput';

describe('SearchInput Component', () => {
  let onSearchMock: ReturnType<typeof vi.fn>;

  beforeEach(() => {
    onSearchMock = vi.fn();
  });

  afterEach(() => {
    cleanup();
    vi.clearAllMocks();
  });

  test('renders the search input with correct snapShot', () => {
    const { asFragment } = render(
      <SearchInput searchText="AAPL" onSearch={() => {}} />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test('calls onSearch when typing in the input field', () => {
    const { getByTestId } = render(
      <SearchInput searchText="" onSearch={onSearchMock} />
    );
    
    const input = getByTestId('search-input');
    
    fireEvent.change(input, { target: { value: 'AAPL' } });

    expect(onSearchMock).toHaveBeenCalledTimes(1);
    expect(onSearchMock).toHaveBeenCalledWith('AAPL');
  });

  test('does not call onSearch when input is not changed', () => {
    const { getByTestId } = render(
      <SearchInput searchText="AAPL" onSearch={onSearchMock} />
    );
    
    const input = getByTestId('search-input');
    fireEvent.change(input, { target: { value: 'AAPL' } });
    
    expect(onSearchMock).not.toHaveBeenCalled();
  });
});
