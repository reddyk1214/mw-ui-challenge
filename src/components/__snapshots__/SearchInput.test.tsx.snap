// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SearchInput Component > renders the search input with correct snapShot 1`] = `
<DocumentFragment>
  <div
    class="SearchInputContainer"
  >
    <div
      class="SearchIconContainer"
    >
      <img
        alt="Search Icon"
        class="SearchIcon"
        src="/src/icons/search_16px.svg"
      />
    </div>
    <input
      aria-label="Search positions"
      class="SearchInput"
      data-testid="search-input"
      placeholder="Search"
      type="search"
      value="AAPL"
    />
  </div>
</DocumentFragment>
`;
